pycqed/measurement/measurement_control.py

```diff
@@ -276,15 +291,16 @@ def measure(self):
                             start_idx:start_idx+self.xlen, 0])
                     self.measure_hard()
         else:
-            raise Exception('Sweep and Detector functions not '
-                            + 'of the same type. \nAborting measurement')
             print(self.sweep_function.sweep_control)
             print(self.detector_function.detector_control)
+            raise Exception('Sweep and Detector functions not '
+                            + 'of the same type. \nAborting measurement')
 
         self.check_keyboard_interrupt()
         self.update_instrument_monitor()
         self.update_plotmon(force_update=True)
-        if self.mode == '2D':
+        if self.mode == '2D' and self.detector_function.detector_control != \
+                'hard':
```

Comment:
I do not know about this but it didn't seem to cause problems.

Reply:
Unfortunately, I do not remember whether I indeed found a measurement for which it was an issue. But if self.detector_function.detector_control == 'hard' in 2D mode, the function update_plotmon_2D_hard is called inside measure_hard, and we should not call update_plotmon_2D. I will add a comment.